### PR TITLE
create configurable topN per workspace

### LINKED ISSUE #542: [FEAT] Control # of context snippets returned
As a user who can edit workspaces, I should be able to increase the amount of `top n` vector results to be returned. The default system-wide is `4` currently.

Now that context windows are no longer an issue we should allow this to be configurable.

This should be an option _per workspace_

### PR DESCRIPTION
**Pull request by shatfield4**
### Pull Request Type

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] 📝 docs

### Relevant Issues

resolves #542 

### What is in this change?

Describe the changes in this PR that are impactful to the repo.

- Add the ability to control the amount of snippets returned by the vector database per workspace (topN value)

### Additional Information

Add any other context about the Pull Request here that was not captured above.

### Developer Validations

- [x] I ran `yarn lint` from the root of the repo & committed changes
- [x] Relevant documentation has been updated
- [x] I have tested my code functionality
- [x] Docker build succeeds locally

### CODE REVIEW

**Comment by shatfield4** [comment]
I still need to QA this before merging.

File: server/utils/vectorDbProviders/chroma/index.js [hunk 1]
```diff
@@ -67,7 +67,8 @@ const Chroma = {
     client,
     namespace,
     queryVector,
-    similarityThreshold = 0.25
+    similarityThreshold = 0.25,
+    topN
```
Comment (timothycarambat) [inline server/utils/vectorDbProviders/chroma/index.js:71]: Why no fallback for this?

File: server/utils/vectorDbProviders/chroma/index.js [hunk 2]
```diff
@@ -272,6 +273,7 @@ const Chroma = {
     input = "",
     LLMConnector = null,
     similarityThreshold = 0.25,
+    topN = null,
```
Comment (timothycarambat) [inline server/utils/vectorDbProviders/chroma/index.js:276]: fallback is null here and 4 in all other providers

File: server/utils/vectorDbProviders/chroma/index.js [hunk 3]
```diff
@@ -290,7 +292,8 @@ const Chroma = {
       client,
       namespace,
       queryVector,
-      similarityThreshold
+      similarityThreshold,
+      topN || 4
```
Comment (timothycarambat) [inline server/utils/vectorDbProviders/chroma/index.js:296]: If we are going to fallback to 4 anyway - just have that be the fallback in the function arguments

**Pull request merged**